### DISCUSSION TITLE
Why doesn't useQueryCache().fetchQuery() use config from the query cache provider?

### DISCUSSION THREAD

**Discussion by janovekj** [TanStack/query#1191] [Q&A]
Given an app wrapped in the following provider:
```
const cache = new QueryCache({
  defaultConfig: { queries: { staleTime: Infinity, cacheTime: Infinity } }
})
<ReactQueryCacheProvider queryCache={queryCache} />
```
Why does every button click trigger a new network call?
```tsx
const cache = useQueryCache();

return (
  <button
    onClick={() =>
      cache.fetchQuery("someKey", () => fetch("/stuff"))
    }
  >
    Fetch
  </button>
);
```
Meanwhile, if I explicitly add a config to the `fetchQuery`, it will respect it, and only fetch once, no matter how many clicks:
```tsx
const cache = useQueryCache();

return (
  <button
    onClick={() =>
      cache.fetchQuery("someKey", () => fetch("/stuff"), {
        staleTime: Infinity,
        cacheTime: Infinity,
      })
    }
  >
    Fetch
  </button>
);
```

I assume this is a conscious design decision, I'm just curious as to why :)

**Comment by TkDodo** [comment]
I think this is a bug that has been fixed with v3. Could you try the v3 beta version?

**Reply by janovekj** [reply]
Ah, yeah, that works perfectly! :) Thanks. Btw, do you know if there is a WIP changelog for v3?

**Reply by TkDodo** [reply]
https://react-query-beta.tanstack.com/guides/migrating-to-react-query-3
